frontend/src/app/passwordReset/MfaSelect/MfaSelect.tsx

```diff
@@ -0,0 +1,119 @@
+import { useEffect, useState } from "react";
+
+import { Card } from "../../commonComponents/Card/Card";
+import { CardBackground } from "../../commonComponents/CardBackground/CardBackground";
+import TextInput from "../../commonComponents/TextInput";
+import Button from "../../commonComponents/Button/Button";
+import RadioGroup from "../../commonComponents/RadioGroup";
+import { emailIsValid } from "../../utils/email";
+
+export const MfaSelect = () => {
+  const [emailAddress, setEmailAddress] = useState("");
+  const [emailError, setEmailError] = useState("");
+  const [formIsDirty, setFormIsDirty] = useState(false);
+  const [mfaOption, setMfaOption] = useState("");
+  const [mfaOptionError, setMfaOptionError] = useState("");
+
+  const validateEmail = () => {
+    setFormIsDirty(true);
+    if (!emailAddress) {
+      setEmailError("Enter your email address");
+      return;
+    }
+    let valid;
+    try {
+      valid = emailIsValid(emailAddress);
+    } catch (e) {
+      valid = false;
+    }
+    if (!valid) {
+      setEmailError("Enter a valid email address");
+    } else {
+      setEmailError("");
+    }
+  };
+
+  const handleSubmit = () => {
+    validateEmail();
+  };
+
+  useEffect(() => {
+    if (formIsDirty) {
+      validateEmail();
+    }
+  });
+
+  const validateMfaOption = () => {
+    if (mfaOption === "") {
+      setMfaOptionError("Select an authentication option");
+    } else {
+      setMfaOptionError("");
+    }
+  };
+
+  return (
+    <CardBackground>
+      <Card logo>
+        <h1 className="font-ui-sm margin-top-3">Reset your password</h1>
+        <p>Confirm your email address to reset your password.</p>
+        <TextInput
+          label={"Email address"}
+          name={"email"}
+          type={"email"}
+          value={emailAddress}
+          errorMessage={emailError}
+          validationStatus={emailError ? "error" : undefined}
+          onBlur={validateEmail}
+          onChange={(evt) => setEmailAddress(evt.target.value)}
+        />
+        <RadioGroup
+          legend="Select your identity verification method"
+          buttons={[
+            {
+              label: "Text message (SMS)",
+              value: "sms",
+            },
+            {
+              label: "Okta Verify",
+              value: "okta",
+            },
+            {
+              label: "Google authenticator",
+              value: "google",
+            },
+            {
+              label: "Security key or biometric authentication",
+              value: "key",
+            },
+            {
+              label: "Phone call",
+              value: "phone",
+            },
+            {
+              label: "Email",
+              value: "email",
+            },
+          ]}
+          selectedRadio={mfaOption}
+          errorMessage={mfaOptionError}
+          validationStatus={mfaOptionError ? "error" : undefined}
+          onBlur={validateMfaOption}
```

Comment:
Error validation doesn't work with onBlur

Reply:
IMO, the best way is to put validation to the `handleSubmit` action., as you do it with email.

Reply:
And if you play with focus by `Tab` key it's work nice.

Reply:
https://user-images.githubusercontent.com/7571965/119817410-9e2e5180-bef6-11eb-9a91-e934bedce4cd.mov